main/ejml-core/src/org/ejml/concurrency/EjmlConcurrency.java

```diff
@@ -45,4 +49,26 @@ public static void setMaxThreads( int maxThreads ) {
 	public static boolean isUseConcurrent() {
 		return USE_CONCURRENT;
 	}
+
+	/**
+	 * Returns true if the operation on the matrix should use the concurrent implementation. If the matrix
+	 * is too small it should always use a single threaded implementation since the overhead will slow it down.
+	 */
+	public static boolean useConcurrent( MatrixSparse mat ) {
+		if (!USE_CONCURRENT)
+			return false;
+
+		return mat.getNonZeroLength() > ELEMENT_THRESHOLD;
```

Comment:
NIT: could write it as `return USE_CONCURRENT && (mat.getNonZeroLength() > ELEMENT_THRESHOLD)`. Short-circuit evaluation assures the same effect and its more compact.

(the same applies for the other method below)

Reply:
reasonable nit. Coding standards at my previous company. It should also be >= I think too.